tools/actions_local_runner.py

```diff
@@ -142,83 +254,76 @@ def grab_specific_steps(steps_to_grab: List[str], job: Dict[str, Any]) -> List[D
     quiet = not args.no_quiet
 
     if args.changed_only:
-        changed_files: Optional[List[str]] = None
-        try:
-            changed_files = find_changed_files()
-        except Exception:
-            # If the git commands failed for some reason, bail out and use the whole list
-            print(
-                "Could not query git for changed files, falling back to testing all files instead",
-                file=sys.stderr
+        relevant_files = relevant_changed_files(args.file_filter)
+
+    if args.file is None:
+        # If there is no .yml file provided, fall back to the list of known
+        # jobs. We use this for flake8 and mypy since they run different
+        # locally than in CI due to 'make quicklint'
+        if args.job not in ad_hoc_steps:
+            raise RuntimeError(
+                f"Job {args.job} not found and no .yml file was provided"
             )
-        if changed_files is not None:
-            relevant_files = []
-            for f in changed_files:
-                for file_filter in args.file_filter:
-                    if f.endswith(file_filter):
-                        relevant_files.append(f)
-                        break
-
-    if args.step is None and args.all_steps_after is None:
-        raise RuntimeError("1+ --steps or --all-steps-after must be provided")
-
-    if args.step is not None and args.all_steps_after is not None:
-        raise RuntimeError("Only one of --step and --all-steps-after can be used")
+        future = ad_hoc_steps[args.job](relevant_files, quiet)
+    else:
+        if args.step is None:
+            raise RuntimeError("1+ --steps must be provided")
 
-    action = yaml.safe_load(open(args.file, "r"))
-    if "jobs" not in action:
-        raise RuntimeError(f"top level key 'jobs' not found in {args.file}")
-    jobs = action["jobs"]
+        action = yaml.safe_load(open(args.file, "r"))
+        if "jobs" not in action:
+            raise RuntimeError(f"top level key 'jobs' not found in {args.file}")
+        jobs = action["jobs"]
 
-    if args.job not in jobs:
-        raise RuntimeError(f"job '{args.job}' not found in {args.file}")
+        if args.job not in jobs:
+            raise RuntimeError(f"job '{args.job}' not found in {args.file}")
 
-    job = jobs[args.job]
+        job = jobs[args.job]
 
-    if args.step is not None:
+        # Pull the relevant sections out of the provided .yml file and run them
         relevant_steps = grab_specific_steps(args.step, job)
-    else:
-        relevant_steps = grab_all_steps_after(args.all_steps_after, job)
+        future = run_steps(relevant_steps, args.job, relevant_files, quiet)
 
-    if sys.version_info > (3, 7):
+    if sys.version_info >= (3, 8):
         loop = asyncio.get_event_loop()
-        loop.run_until_complete(run_steps(relevant_steps, args.job, relevant_files, quiet))
+        loop.run_until_complete(future)
     else:
         raise RuntimeError("Only Python >3.7 is supported")
```

Comment:
since you've already changed the above `if` statement to use `>=` instead of `>`:
```suggestion
        raise RuntimeError("Only Python >=3.8 is supported")
```